test/functional/test_framework/util.py

```diff
@@ -385,6 +385,17 @@ def set_node_times(nodes, t):
     for node in nodes:
         node.setmocktime(t)
 
+def disconnect_nodes(from_connection, node_num):
+    timeout = 5
+    while True:
+        peer_ids = [peer['id'] for peer in from_connection.getpeerinfo() if "testnode%d" % node_num in peer['subver']]
+        if not peer_ids:
+            break
+        [from_connection.disconnectnode(nodeid=peer_id) for peer_id in peer_ids]
```

Comment:
It *should* only need to be called once, but I called this every loop in case either:

- `disconnectnode` fails for one of the nodes and we need to rerun it.
- a new connection to the node is added while we're in the loop.

I don't think either of those should happen, so we can probably just remove this.